### PR TITLE
GH-39037: [Java] Remove (Contrib/Experimental) mention in Flight SQL

### LINKED ISSUE #39037: [Java] Remove Contrib/Experimental from Flight SQL
### Describe the enhancement requested

Remove the `(Contrib/Experimental)` mention in Java Flight SQL modules which also show up in Maven Central website, as those modules have been part of the project for a while now.

### Component(s)

Java

### PR DESCRIPTION
**Pull request by laurentgo**
### Rationale for this change

Considering that Flight SQL has been present for a while and should be fairly stable, remove the `(Contrib/Experimental)` mention from the pom file which also shows up on Maven Central UI pages

### Are these changes tested?

Local test but there's no code change, only cosmetic

### Are there any user-facing changes?

None

* Closes: #39037

### CODE REVIEW

**Comment by lidavidm** [comment]
We still declare it as experimental in the format itself, though.

File: cpp/submodules/parquet-testing
Comment (mapleFU) [inline cpp/submodules/parquet-testing:1]: Oops seems we shouldn't edit `parquet-testing`?

Comment (laurentgo) [reply cpp/submodules/parquet-testing:1]: Yes, was not supposed to. Thanks for pointing it out

**Review by lidavidm — COMMENTED** [review]
LGTM once we remove the submodule change

**Comment by laurentgo** [comment]
@lidavidm Would you prefer I also take care of the rest of the mentions, following up on the vote on the mailing list?

**Comment by lidavidm** [comment]
up to you, I already filed a new issue for that so don't feel like you have to here

**Comment by vibhatha** [comment]
Also took a look and LGTM.

**Comment by laurentgo** [comment]
> up to you, I already filed a new issue for that so don't feel like you have to here

Ok, will keep my change as-is then. Thanks

**Comment by vibhatha** [comment]
A CI is failing, but it seems unrelated.

**Pull request merged**